### DISCUSSION TITLE
Why does the world not save and how can i load it with the correct Light?

### DISCUSSION THREAD

**Discussion by Jochengehtab** [Minestom/Minestom#1780] [Help]
Hello i tried to save and load a world but it doesnt word. I am using the SOTA Extension for Lightning.

These are my Trys:
(Save the World)
`

InstanceManager instanceManager = Main.getInstanceManager();

            DimensionType lobby = DimensionType.builder(NamespaceID.from("minestom:lobby"))
                    .ambientLight(2.0f)
                    .build();
            MinecraftServer.getDimensionTypeManager().addDimension(lobby);

            InstanceContainer instance = new SOTAInstance (UUID.randomUUID(), DimensionType.OVERWORLD);

            Manager.set("./ProtectorPos.json", "lobby", String.valueOf(instance.getUniqueId()));
            instance.setGenerator(unit -> unit.modifier().fillHeight(0, 2, Block.STONE));
            Tag<String> tag = Tag.String("lobby");
            instance.setTag(tag, "lobby");
            instanceManager.registerInstance(instance);

            instance.saveChunksToStorage();`

and this to load the world

`

InstanceManager instanceManager = Main.getInstanceManager();
   
        JSONObject jsonObject = Manager.getHoleFile("./ProtectorPos.json");
            SOTAChunkLoader sotaChunkLoader = new SOTAChunkLoader(Path.of("./world"));
            SOTAInstance instance = (SOTAInstance) instanceManager.getInstance (UUID.fromString((String) jsonObject.get("lobby")));
            instance.setChunkLoader(sotaChunkLoader);

            Player player = (Player) sender;
            player.sendMessage(Component.text("You are now in the lobby", TextColor.color(51, 204, 51)));

            player.setInstance(Objects.requireNonNull(instance), new Pos(5, 5, 5));`

I hope someone can help me because i try that for days and it so still not work.
NOTE: I have already check the UUID.  The UUID is still always the correct UUID.

**Comment by Tofaa2** [comment]
I'd suggest contacting the people who made the "SOTA Extension"

**Reply by Jochengehtab** [reply]
No, i think it is not a Problem with the SOTA because for example SOATInstance extends fromm Instance.

**Reply by Tofaa2** [reply]
That doesnt mean its not the problem.

**Comment by Jochengehtab** [comment]
I have solve it
